File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        string res = "";
        for(int l = 0; l < s.size(); l++){
            int i = l;
            int j = l;
            while(i >= 0 && j < s.size() && s[i] == s[j]){
                if(res.size() < j - i + 1){
                    res = s.substr(i, j - i + 1);
                }
                i--;
                j++;
            }

            i = l;
            j = l + 1;
            while(i >= 0 && j < s.size() && s[i] == s[j]){
                if(res.size() < j - i + 1){
                    res = s.substr(i, j - i + 1);
                }
                i--;
                j++;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        # map = {[26] : [s1, s2, s3]}
        anas = {}
        for s in strs:
            alpha = [0 for _ in range(26)]
            for c in s:
                alpha[ord(c) - ord('a')] += 1
            if "".join(map(str, alpha)) in anas:
                anas["".join(map(str, alpha))].append(s)
            else:
                anas["".join(map(str, alpha))] = [s]
        res = []
        for key,val in anas.items():
            res.append(val)
        return res